src/libraries/System.Private.Uri/src/System/UriExt.cs

```diff
@@ -593,6 +599,11 @@ private Uri(Flags flags, UriParser? uriParser, string uri)
             _flags = flags;
             _syntax = uriParser!;
             _string = uri;
+
+            if (uriParser is null)
+            {
+                DebugSetLeftCtor();
```

Comment:
Why only if uriParser is null?

Reply:
uriParser being null means that the uri is a relative Uri. It is fully initialized after a call to this ctor.

Absolute Uris are initialized later with a call to InitializeUri on the newly created instance.

Added a comment.